demos/embeddings/deep-graph-infomax-cora.ipynb

```diff
@@ -6,7 +6,7 @@
    "source": [
```

Comment:
<p>Having this be conditional means that we're not testing the notebook with each of the algorithms, only whatever is hard coded, and requires extra conditionals to get the right generator (etc.). In this particular case, the latter isn't too bad, but it is very slightly confusing (i.e. a user reading this would have to "decode" it to be sure they're doing the right thing).</p><p> </p><p>An alternative would be to have this notebook do all/most of the algorithms directly, as a comparison of their performance. It could potentially be framed something like</p><p> </p><ul><li>teach how to use DGI with one algorithm (GCN for simplicity, or whichever gives the best results) with all of the explanations, and demonstrate the downstream task(s) like node classification and TSNE</li><li>run just the DGI -&gt; node classification training task for all of the other algorithms, only explaining whatever differences (since there's less need for explanation, the cells could be larger, with less intermediate output, so the notebook doesn't explode in size).</li><li>Give a comparison summary table/dataframe of the results for each of them at the end as a conclusion</li></ul><p> </p><p>What do you think?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
yeah I think you're right that this might be confusing - especially for new users. I'll make the first section just about one algo, and then do a comparison summary to demonstrate that DGI can be used for a lot of different algos 👍

Reply:
I've added this and the results for APPNP and GAT are 30% and 40% respectively which seem a bit low - i'll investigate and try and figure out whats going on 🤔

Reply:
APPNP just applies a MLP and then does nieghbood smoothing, so I think it makes sense that thie doesnt work for APPNP. Not sure about GAT though

Reply:
Could you explain a bit more about your reasoning why that process means APPNP "should" fail? I feel like one-layer GCN like used here is fairly similar conceptually (squashing the MLP and neighbourhood smoothing into a single step) and that seems to go ok?